### ISSUE TITLE
Server-assigned queue names with configurable options [AMQP-345]

### ISSUE THREAD

**Issue by spring-operator** [spring-projects/spring-amqp#1909]
**[Jonathan Castellani](https://jira.spring.io/secure/ViewProfile.jspa?name=jcastellani)** opened **[AMQP-345](https://jira.spring.io/browse/AMQP-345?redirect=false)** and commented

We would like the ability to create RabbitMQ queues with server-assigned names (similar to the no-arg RabbitAdmin.declareQueue() method), but be allowed to create non-exclusive, non-autodelete queues.

The motivation for this is the desire to be able to run integration test suites in parallel against a single broker.  Some of our tests start with an empty queue (which they ensure is empty by draining it) and proceed from there.  Obviously, this tends to interfere with any other tests running at the same time against the broker.  It would be nice to have server-assigned queue names for the test queues, and their associated DLQs.  (DLQing is one of the things we test for.)

We tried doing this by passing an empty string in as the queue name to a call to RabbitAdmin.decalreQueue("", false, false, false, null), but as the method returns void, we cannot get to the server-assigned queue name.

We could certainly create the queues via the RabbitTemplate ourselves, but I could see others benefitting from a new Spring AMQP API call for this.

Many thanks,
JC

---

**Affects:** 1.2.0.GA
